VoiceProcessor.swift

```diff
@@ -69,31 +81,44 @@ public class VoiceProcessor {
         }
         
         AudioQueueStart(queue, nil)
+        started = true
     }
     
     public func stop() {
+        if !started {
+            return
+        }
         guard let audioQueue = audioQueue else {
             return
         }
         AudioQueueFlush(audioQueue)
         AudioQueueStop(audioQueue, true)
         AudioQueueDispose(audioQueue, true)
         audioCallback = nil
+        started = false
     }
     
     private func createAudioQueueCallback() -> AudioQueueInputCallback {
         return { userData, queue, bufferRef, startTimeRef, numPackets, packetDescriptions in
-            
             // `self` is passed in as userData in the audio queue callback.
             guard let userData = userData else {
                 return
             }
+            
             let `self` = Unmanaged<VoiceProcessor>.fromOpaque(userData).takeUnretainedValue()
             
+            guard let frameLength = self.frameLength else {
+                return
+            }
+            
+            guard frameLength == numPackets else {
+                return
```

Comment:
should probably print that something went wrong here. Even better would be to have an optional error callback as an input parameter that is called when something goes wrong.